apps/openmw/switch_startup.cpp

```diff
@@ -0,0 +1,211 @@
+#include <switch.h>
+#include <stdarg.h>
+#include <SDL.h>
+
+#include <components/files/configurationmanager.hpp>
+#include <components/debug/debuglog.hpp>
+// TODO: separate this into a component maybe
+#include <apps/mwiniimporter/importer.hpp> 
+
+#include "switch_startup.hpp"
+
+#ifdef _DEBUG
+
+static int nxlinkSock = -1;
+
+extern "C" void userAppInit(void)
+{
+    if (R_SUCCEEDED(socketInitializeDefault()))
+        nxlinkSock = nxlinkStdio();
+}
+
+extern "C" void userAppExit(void)
+{
+    if (nxlinkSock >= 0)
+        close(nxlinkSock);
+    socketExit();
+}
+
+#endif
+
+namespace Switch
+{
+    std::string username = "";
+
+    // read username from the account with the specified id
+    void readUsername(u128 userId)
+    {
+        auto rc = accountInitialize();
+        if (R_FAILED(rc))
+            return;
+
+        AccountProfile profile;
+        rc = accountGetProfile(&profile, userId);
+
+        if (R_SUCCEEDED(rc))
+        {
+            AccountProfileBase pb;
+            rc = accountProfileGet(&profile, nullptr, &pb);
+
+            if (R_SUCCEEDED(rc))
+            {
+                // HACK: check if the username is printable ascii
+                // otherwise throw it out for the time being, since fs seems
+                // allergic to certain UTF-8 characters 

```

Comment:
Google says that Switch uses the exFAT filesystem with UTF-16 encoding, not UTF-8. Probably the same thing for accounts. In this case you should convert your paths to UTF-8 by using `bconv`, as we do on Windows.